src/walletSdk/Anchor/index.ts

```diff
@@ -201,6 +224,183 @@ export class Anchor {
     }
   }
 
+  /**
+  * Watch all transactions returned from a transfer server. When new or
+  * updated transactions come in, run an `onMessage` callback.
+  *
+  * On initial load, it'll return ALL pending transactions via onMessage.
+  * Subsequent messages will be any one of these events:
+  *  * Any new transaction appears
+  *  * Any of the initial pending transactions change any state
+  *
+  * You may also provide an array of transaction ids, `watchlist`, and this
+  * watcher will always react to transactions whose ids are in the watchlist.
+  */
+  watchAllTransactions(params: {
```

Comment:
Can we move implementation to a separate class? And here just make a delegate call (anchor class is getting quite big)
(Maybe even make sense to move the method too, so it will be `anchor.watcher.watchAllTransaction()`
If we plan to add more functionality to the watcher later.)
Also, if we have a watcher class, they will be able to create multiple watchers

Reply:
I'll try moving to a separate class

Reply:
@Ifropc just moved it to a `Watcher` class, I'll work on updating the [other PR](https://github.com/stellar/typescript-wallet-sdk/pull/14) now